src/tests/Interop/Interop.csproj

```diff
@@ -21,6 +21,8 @@
     <Compile Include="LayoutClass/*.cs" />
     <Compile Include="ArrayMarshalling/**/*.cs" />
     <Compile Include="MarshalAPI/**/*.cs" />
+    <!-- Disable build of GenericFunctionPointer tests on mono -->
+    <Compile Condition="'$(RuntimeFlavor)' == 'mono'" Remove="MarshalAPI/FunctionPointer/GenericFunctionPointer.cs" />
```

Comment:
I think that the conditional compilation like this is violating invariants of the test builds. From https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/coreclr/test-configuration.md#adding-test-guidelines: "Therefore the managed portion of each test must not contain - Target platform dependent conditionally included files"

The pattern to disable tests for Mono is `<CLRTestTargetUnsupported Condition="'$(RuntimeFlavor)' == 'mono'">true</CLRTestTargetUnsupported>`.

Reply:
We do build the tests separately for Mono and CoreCLR builds (and we depend on this in the test source generator today).

I think this is fine for now (and is better than splitting out these tests into a separate process just to avoid running these tests on Mono).